opentelemetry-stdout/src/trace/transform.rs

```diff
@@ -0,0 +1,235 @@
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ResourceSpans {
+    resource: Resource,
+    scope_spans: Vec<ScopeSpans>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    schema_url: Option<Cow<'static, str>>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ScopeSpans {
+    scope: Scope,
+    spans: Vec<Span>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    schema_url: Option<Cow<'static, str>>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Span {
+    trace_id: String,
+    span_id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    trace_state: Option<String>,
+    parent_span_id: String,
+    name: Cow<'static, str>,
+    kind: SpanKind,
+    #[serde(serialize_with = "as_unix_nano")]
+    start_time_unix_nano: SystemTime,
+    #[serde(serialize_with = "as_unix_nano")]
+    end_time_unix_nano: SystemTime,
+    attributes: Vec<KeyValue>,
+    dropped_attributes_count: u32,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    events: Vec<Event>,
+    dropped_events_count: u32,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    links: Vec<Link>,
+    dropped_links_count: u32,
+    status: Status,
+}
+
+fn as_unix_nano<S>(time: &SystemTime, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let nanos = time
+        .duration_since(UNIX_EPOCH)
+        .unwrap_or_default()
+        .as_nanos();
+
+    serializer.serialize_u128(nanos)
+}
+
+impl From<opentelemetry_sdk::export::trace::SpanData> for Span {
+    fn from(value: opentelemetry_sdk::export::trace::SpanData) -> Self {
+        Span {
+            trace_id: format!("{:x}", value.span_context.trace_id()),
+            span_id: format!("{:x}", value.span_context.span_id()),
+            trace_state: Some(value.span_context.trace_state().header()).filter(|s| !s.is_empty()),
+            parent_span_id: Some(format!("{:x}", value.parent_span_id))
+                .filter(|s| s != "0")
+                .unwrap_or_default(),
+            name: value.name,
+            kind: value.span_kind.into(),
+            start_time_unix_nano: value.start_time,
+            end_time_unix_nano: value.end_time,
+            dropped_attributes_count: value.attributes.dropped_count(),
+            attributes: value.attributes.into_iter().map(Into::into).collect(),
+            dropped_events_count: value.events.dropped_count(),
+            events: value.events.into_iter().map(Into::into).collect(),
+            dropped_links_count: value.links.dropped_count(),
+            links: value.links.into_iter().map(Into::into).collect(),
+            status: value.status.into(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+enum SpanKind {
+    #[allow(dead_code)]
+    Unspecified = 0,
+    Internal = 1,
+    Server = 2,
+    Client = 3,
+    Producer = 4,
+    #[allow(dead_code)]
```

Comment:
Consumer requires allow(dead_code) ?

Reply:
Oh good catch, this was from earlier but is not necessary.